File: week-02/day-01/32_draw_diagonal.py

# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was

num = input("Enter a number: ")
num = int(num)
str = "%" * num
within_space = num - 2
space = " " * num

print(str)

for i in range(within_space):
    print(str[0] + space[:i] + str[0] + space[:num - i - 3] + str[0])
print(str)

File: week-07/day-01/32_draw_diagonal.js
'use strict';

let lineCount = 20;
let squareChar = '%';

// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is

console.log(squareChar.repeat(lineCount));
for (let i = 0; i < lineCount - 2; i++) {
    console.log(squareChar + ' '.repeat(i) + squareChar + ' '.repeat( lineCount - i - 3) + squareChar);
}
console.log(squareChar.repeat(lineCount));
